src/script-events.c

```diff
@@ -0,0 +1,177 @@
+#include <linux/kconfig.h>
+#include <uapi/linux/ptrace.h>
+#include <linux/binfmts.h>
+#include <linux/dcache.h>
+#include "types.h"
+#include "offsets.h"
+#include "common.h"
+
+struct bpf_map_def SEC("maps/load_script_map") load_script_map = {
+    .type = BPF_MAP_TYPE_HASH,
+    .key_size = sizeof(u32),
+    .value_size = sizeof(size_t),
+    .max_entries = 1024,
+    .pinning = 0,
+    .namespace = "",
+};
+
+struct bpf_map_def SEC("maps/script_events") script_events = {
+    .type = BPF_MAP_TYPE_PERF_EVENT_ARRAY,
+    .key_size = sizeof(u32),
+    .value_size = sizeof(u32),
+    .max_entries = MAX_TELEMETRY_STACK_ENTRIES * 64,
+    .pinning = 0,
+    .namespace = "",
+};
+
+SEC("kprobe/script_load")
+int kprobe__script_load(struct pt_regs *ctx)
+{
+    struct linux_binprm *bprm = (struct linux_binprm *)PT_REGS_PARM1(ctx);
+    u32 index = (u32)bpf_get_current_pid_tgid();
+
+    bpf_map_update_elem(&load_script_map, &index, &bprm, BPF_ANY);
+
+    return 0;
+}
+
+SEC("kretprobe/ret_script_load")
+int kretprobe__ret_script_load(struct pt_regs *ctx)
+{
+    char br = 0;
```

Comment:
I honestly can't remember what was going through my head when I named this variable.